### PR TITLE
esp32/boards: Add board files for the Olimex ESP32 PoE boards.

### PR DESCRIPTION
**Pull request by robert-hh**
The important setting is about the PHY interface configuration. Uisng dynamic configuration of the ref_clk interface as suggested in PR #10329 is an alternative to the static configuration, but it requires esp-idf v4.4. As of now, the ESP32 images are built with esp_idf v4.2.2.

The configuration matches the Olimex ESP32 Gateway as well.

### CODE REVIEW

**Comment by dpgeorge** [comment]
I think the .html file was added in error.  Can it be removed?

File: ports/esp32/boards/OLIMEX_ESP32_POE/sdkconfig.board [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CONFIG_ETH_PHY_INTERFACE_RMII=y
+CONFIG_ETH_RMII_CLK_OUTPUT=y
+CONFIG_ETH_RMII_CLK_OUT_GPIO=17
+CONFIG_LWIP_LOCAL_HOSTNAME="ESP32_POE"
+
+CONFIG_COMPILER_OPTIMIZATION_SIZE=y
```
[the remark this answers is not in the document]
Comment (robert-hh) [reply ports/esp32/boards/OLIMEX_ESP32_POE/sdkconfig.board:6]: Sure.

[the remark this answers is not in the document]
Comment (robert-hh) [reply ports/esp32/boards/OLIMEX_ESP32_POE/sdkconfig.board]: Changed, tested and force pushed to the olimex_esp32_poe branch.

**Comment by robert-hh** [comment]
Thank you. The PR for the picture exists as well.

**Comment by robert-hh** [comment]
@dpgeorge There is an inconsistency with the Olimex esp32 firmware images on the web site. The picture for the board exists, and the text refers for the Olimex boards, but  the firmware is linked to the generic ESP32. But  that will work only with firmware with esp-idf v4.4.x. The esp-idf v4.2.x firmware does not support the ref_clk options, which would be required when using the generic firmware. I know it's quite a mess with all the different boards & esp-idf versions. I searched for the `build_esp32_boards` script, but did not find it.

File: ports/esp32/boards/OLIMEX_ESP32_POE/board.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "features": [
+        "BLE",
+        "WiFi",
+        "MicroSD",
+        "Battery Charging",
+        "Ethernet",
+        "PoE",
+        "Breadboard friendly"
+    ],
+    "id": "esp32",
```
Comment (dpgeorge) [inline ports/esp32/boards/OLIMEX_ESP32_POE/board.json:15]: @robert-hh this line should not be here, it's making it link to the generic esp32 firmware

File: ports/esp32/boards/OLIMEX_ESP32_POE/board.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "images": [
+        "ESP32-POE-ISO-1.jpg"
+    ],
+    "mcu": "esp32",
+    "product": "ESP32",
```
Comment (dpgeorge) [inline ports/esp32/boards/OLIMEX_ESP32_POE/board.json:20]: @robert-hh this line should be more descriptive, eg "Olimex ESP32 POE".  The download page will use this as the title.

File: ports/esp32/boards/OLIMEX_ESP32_POE/board.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    "thumbnail": "",
+    "url": "https://www.olimex.com/",
+    "vendor": "Olimex"
```
Comment (dpgeorge) [inline ports/esp32/boards/OLIMEX_ESP32_POE/board.json:23]: @robert-hh I see some other boards have this as all uppercase, ie OLIMEX.  Which is correct?

Comment (robert-hh) [reply ports/esp32/boards/OLIMEX_ESP32_POE/board.json:23]: The vendor calls itself on the `About` page of it's web site both `Olimex Ltd` and `OLIMEX Ltd.`, so I'll change it to OLIMEX for consistency.

File: ports/esp32/boards/OLIMEX_ESP32_POE/board.json (hunk 2 above)
Comment (robert-hh) [reply ports/esp32/boards/OLIMEX_ESP32_POE/board.json:15]: Is the `id` tag required?

File: ports/esp32/boards/OLIMEX_ESP32_POE/board.json (hunk 3 above)
Comment (robert-hh) [reply ports/esp32/boards/OLIMEX_ESP32_POE/board.json:20]: Done

**Comment by robert-hh** [comment]
I'll make a new PR, since I have already deleted the previous branch.

**Pull request merged**